packages/compass-global-writes/src/store/reducer.ts

```diff
@@ -625,7 +625,8 @@ export const createShardKey = (
 
 // Exporting this for test only to stub it and set
 // its value. This enables to test cancelShardingaction.
-export const showConfirmation = showConfirmationModal;
+export const showConfirmation: typeof showConfirmationModal =
```

Comment:
I'm curious as to why this is needed? 🤔 It seems redundant to me.
Same with the change in `packages/compass-indexes/src/modules/regular-indexes.ts`

Reply:
ts started to complain about this
`The inferred type of 'showConfirmation' cannot be named without a reference to '@mongodb-js/compass-components/node_modules/@leafygreen-ui/confirmation-modal'. This is likely not portable. A type annotation is necessary.ts(2742)`.

I'll close this PR as I split the removal of combobox from this.